src/main/java/com/microsoft/jfr/Recording.java

```diff
@@ -0,0 +1,226 @@
+         * The {@code Recording} has been created.
+         */
+        NEW,
+        /**
+         * The {@code Recording} has been started.
+         */
+        RECORDING,
+        /**
+         * The {@code Recording} has been stopped.
+         */
+        STOPPED,
+        /**
+         * The {@code Recording} has been closed. Once the
+         * recording has been closed, it is no longer usable.
+         */
+        CLOSED;
+    }
+
+    // Format for IllegalStateException that this class might throw
+    private final static MessageFormat illegalStateFormat = new MessageFormat("Recording state {0} not in [{1}]");
+    private static String createIllegalStateExceptionMessage(State actual, State expected, State... others) {
+        String[] args = new String[]{actual.name(), expected.name()};
+        if (others != null) {
+            for (State state : others) args[1] = args[1].concat(", ").concat(state.name());
+        }
+        String  msg = illegalStateFormat.format(args);
+        return msg;
+    }
+
+    final private FlightRecorderConnection connection;
+    final private RecordingOptions recordingOptions;
+    final private RecordingConfiguration recordingConfiguration;
+
+    private volatile long id = -1;
+    private final AtomicReference<State> state;
+
+    /**
+     * Create a {@code Recording}. Recordings are created from
+     * {@link FlightRecorderConnection#newRecording(RecordingOptions, RecordingConfiguration)}
+     * @param connection A connection to the FlightRecorderMXBean on an MBean server
+     * @param recordingOptions The options to be used for the recording
+     * @param recordingConfiguration The settings for events to be collected by the recording
+     */
+    /*package*/ Recording(
+            FlightRecorderConnection connection,
+            RecordingOptions recordingOptions,
+            RecordingConfiguration recordingConfiguration) {
+        this.connection = connection;
+        this.recordingOptions = recordingOptions;
+        this.recordingConfiguration = recordingConfiguration;
+        this.state = new AtomicReference<>(State.NEW);
+    }
+
+    /**
+     * Get the recording id. The recording does not have an id until the recording is started.
+     * @return The recording id, or {@code -1} if the recording was never started.
+     */
+    public long getId() throws IllegalStateException {
+        return id;
+    }
+
+    /**
+     * Start a recording. A recording may not be started after it is closed.
+     * @throws IOException A communication problem occurred when talking to the MBean server.
+     * @throws IllegalStateException This {@code Recording} is closed.
+     * @throws InternalError Wraps a {@code javax.management.JMException}. See {@link InternalError}.
+     * @return The recording id. 
+     */
+    public long start() throws IOException, IllegalStateException {
+        // state machine transitions: NEW -> RECORDING or STOPPED -> RECORDING, otherwise remain in state
```

Comment:
Is this the obvious/ideal place to have the state machine comment?  Ignore this if it makes no sense